packages/ec-core/src/generator/slice_cloning.rs

```diff
@@ -0,0 +1,40 @@
+use rand::{distributions::Slice, prelude::Distribution};
+
+/// Generate a random element from an array of options, cloning the choosen
+/// element.
+#[derive(Debug, Clone, Copy)]
+pub struct SliceCloning<'a, T>(Slice<'a, T>);
```

Comment:
Sure. SliceCloning mirrors `rand::Slice` which is  a Distribution<&'a T>. We can't really do anything else here either since the orphan rules prevent us from just implementing Distribution for slices /vecs/... since the trait is from a foreign crate, so we need some newtype of some kind.